From 83fd182901d50d150416afaa1236c3b798b320e7 Mon Sep 17 00:00:00 2001
From: "Dr. David Alan Gilbert" <dgilbert@redhat.com>
Date: Fri, 14 Dec 2018 17:02:08 +0000
Subject: [PATCH 3/8] pc: PC_RHEL7_6_COMPAT

RH-Author: Dr. David Alan Gilbert <dgilbert@redhat.com>
Message-id: <20181214170211.14267-4-dgilbert@redhat.com>
Patchwork-id: 83513
O-Subject: [RHEL8 qemu-kvm PATCH v3 3/6] pc: PC_RHEL7_6_COMPAT
Bugzilla: 1655820
RH-Acked-by: Laurent Vivier <lvivier@redhat.com>
RH-Acked-by: Cornelia Huck <cohuck@redhat.com>
RH-Acked-by: Thomas Huth <thuth@redhat.com>

From: "Dr. David Alan Gilbert" <dgilbert@redhat.com>

Add the PC_RHEL7_6_COMPAT macro derived from the 3.0 and 2.12
upstream macros.

Signed-off-by: Dr. David Alan Gilbert <dgilbert@redhat.com>
Signed-off-by: Danilo C. L. de Paula <ddepaula@redhat.com>
---
 include/hw/i386/pc.h | 19 +++++++++++++++++++
 1 file changed, 19 insertions(+)

diff --git a/include/hw/i386/pc.h b/include/hw/i386/pc.h
index efcb208..5533863 100644
--- a/include/hw/i386/pc.h
+++ b/include/hw/i386/pc.h
@@ -998,6 +998,25 @@ extern void igd_passthrough_isa_bridge_create(PCIBus *bus, uint16_t gpu_dev_id);
             .value = "on",\
         },
 
+/* Similar to PC_COMPAT_3_0 + PC_COMPAT_2_12, but:
+ * all of the 2_12 stuff was already in 7.6 from bz 1481253
+ */
+#define PC_RHEL7_6_COMPAT \
+        HW_COMPAT_RHEL7_6 \
+        { /* PC_RHEL7_6_COMPAT from PC_COMPAT_3_0 */ \
+            .driver   = TYPE_X86_CPU,\
+            .property = "x-hv-synic-kvm-only",\
+            .value    = "on",\
+        },{ /* PC_RHEL7_6_COMPAT from PC_COMPAT_3_0 */ \
+            .driver   = "Skylake-Server" "-" TYPE_X86_CPU,\
+            .property = "pku",\
+            .value    = "off",\
+        },{ /* PC_RHEL7_6_COMPAT from PC_COMPAT_3_0 */ \
+            .driver   = "Skylake-Server-IBRS" "-" TYPE_X86_CPU,\
+            .property = "pku",\
+            .value    = "off",\
+        },
+
 /* Similar to PC_COMPAT_2_11 + PC_COMPAT_2_10, but:
  * - x-hv-max-vps was backported to 7.5
  * - x-pci-hole64-fix was backported to 7.5
-- 
1.8.3.1

